pkg/sql/row/errors.go

```diff
@@ -122,61 +119,68 @@ func NewLockNotAvailableError(
 			"could not obtain lock on row in interleaved table")
 	}
 
-	index, datums, err := decodeRowInfo(ctx, tableDesc, key, nil)
+	index, colNames, values, err := DecodeRowInfo(ctx, tableDesc, key, nil, false)
 	if err != nil {
 		return pgerror.Newf(pgcode.LockNotAvailable,
 			"could not obtain lock on row: decoding err=%s", err)
 	}
 
-	datumStrs := make([]string, len(datums))
-	for i := range datums {
-		datumStrs[i] = datums[i].String()
-	}
 	return pgerror.Newf(pgcode.LockNotAvailable,
 		"could not obtain lock on row (%s)=(%s) in %s@%s",
-		strings.Join(index.ColumnNames, ","),
-		strings.Join(datumStrs, ","),
+		strings.Join(colNames, ","),
+		strings.Join(values, ","),
 		tableDesc.Name,
 		index.Name)
 }
 
-// decodeRowInfo takes a table descriptor, a key, and an optional value and
+// DecodeRowInfo takes a table descriptor, a key, and an optional value and
 // returns information about the corresponding SQL row. If successful, the index
-// and datums corresponding to the provided key are returned.
-func decodeRowInfo(
+// and corresponding column names and values to the provided KV are returned.
+func DecodeRowInfo(
 	ctx context.Context,
 	tableDesc *sqlbase.ImmutableTableDescriptor,
 	key roachpb.Key,
 	value *roachpb.Value,
-) (*descpb.IndexDescriptor, tree.Datums, error) {
+	allColumns bool,
+) (*descpb.IndexDescriptor, []string, []string, error) {
 	// Strip the tenant prefix and pretend to use the system tenant's SQL codec
 	// for the rest of this function. This is safe because the key is just used
 	// to decode the corresponding datums and never escapes this function.
 	codec := keys.SystemSQLCodec
 	key, _, err := keys.DecodeTenantPrefix(key)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
 	indexID, _, err := sqlbase.DecodeIndexKeyPrefix(codec, tableDesc, key)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
 	index, err := tableDesc.FindIndexByID(indexID)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
 	var rf Fetcher
 
+	colIDs, _ := index.FullColumnIDs()
```

Comment:
might be worth moving this into a utility function on the table descriptor for future use

Reply:
Actually I realized this was a behavior change from before when it was just ColumnIDs so i did it a bit differently